packages/params/src/forkName.ts

```diff
@@ -20,28 +20,70 @@ export enum ForkSeq {
   deneb = 4,
 }
 
+function exclude<T>(coll: T[], val: T[]): T[] {
```

Comment:
since this removes keys in `val` from `coll` shouldn't the function signature be like this
```suggestion
function exclude<T, U extends T>(coll: T[], val: U[]): Exclude<T, U>[]
```
Not sure if we care here, seems to always return `never[]` if I do this, haven't digged into why

Reply:
That is useful only with the tuples, not arrays.

Reply:
`forkAll` is a tuple but we type it as an array

Reply:
`forkAll` is not a tuple, it's an array and we don't type it it's derived from the enum.

Reply:
it's not typed as a tuple in terms of typescript type right now but in the context of Ethereum CL spec, it's a tuple.

The correct typescript type for `forkAll` would be the following:
```ts
type ForkAll = [ForkName.phase0, ForkName.altair, ForkName.bellatrix, ForkName.capella, ForkName.deneb];
```

This would be invalid if typed as tuple
```ts
// Type 'ForkName.altair' is not assignable to type 'ForkName.bellatrix'.ts(2322)
const forkAll: ForkAll = [ForkName.phase0, ForkName.altair, ForkName.altair];
```
But it is allowed if typed as array

Reply:
- see https://github.com/ChainSafe/lodestar/pull/6916 it's easier to show in code